File: docs/snippets/all/line_strip2d_simple.py
"""Log a simple line strip."""

import rerun as rr
import rerun.blueprint as rrb

rr.init("rerun_example_line_strip2d", spawn=True)

rr.log(
    "strip",
    rr.LineStrips2D([[[0, 0], [2, 1], [4, -1], [6, 0]]]),
)

# Set view bounds:
rr.send_blueprint(rrb.Spatial2DView(visual_bounds=rrb.VisualBounds(min=[-1, -3], max=[7, 3])))

File: docs/snippets/all/line_strip2d_simple.cpp
// Log a simple line strip.

#include <rerun.hpp>

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_line_strip2d");
    rec.spawn().exit_on_failure();

    const auto strip = rerun::LineStrip2D({{0.f, 0.f}, {2.f, 1.f}, {4.f, -1.f}, {6.f, 0.f}});
    rec.log("strip", rerun::LineStrips2D(strip));

    // Log an extra rect to set the view bounds
    rec.log("bounds", rerun::Boxes2D::from_centers_and_sizes({{3.0f, 0.0f}}, {{8.0f, 6.0f}}));
}
